asab/library/service.py

```diff
@@ -637,58 +637,93 @@ async def get_item_metadata(self, path: str) -> typing.Optional[dict]:
 		L.info("Item '{}' not found in directory '{}'.".format(filename, directory))
 		return None
 
-
-	async def export(self, path: str = "/", remove_path: bool = False) -> typing.IO:
+	async def export(
+		self,
+		path: str = "/",
+		remove_path: bool = False,
+	) -> typing.IO:
 		"""
-		Return a file-like stream containing a gzipped tar archive of the library contents of the path.
+		Produce a gzipped tar of global and, optionally, tenant layers.
 
 		Args:
-			path: The path to export.
-			tenant (str | None ): The tenant to use for the operation.
-			remove_path: If `True`, the path will be removed from the tar file.
-
+			path: Directory path to export (must end with '/').
+			remove_path: If True, strip `path` prefix from archived names.
+			tenant: If True, append each tenant's layer under `.tenants/<tenant>/`.
 		Returns:
-			A file object containing a gzipped tar archive.
+			A file-like object with the tar.gz archive.
 		"""
-
 		_validate_path_directory(path)
 
+		provider = self.Libraries[0]
 		fileobj = tempfile.TemporaryFile()
 		tarobj = tarfile.open(name=None, mode='w:gz', fileobj=fileobj)
 
-		items = await self._list(path, providers=self.Libraries[:1])
-		recitems = list(items[:])
-
-		while len(recitems) > 0:
-
-			item = recitems.pop(0)
-			if item.type != 'dir':
-				continue
-
-			child_items = await self._list(item.name, providers=item.providers)
-			items.extend(child_items)
-			recitems.extend(child_items)
-
+		# -- Global layer --
+		items = await self._collect_items(path, providers=[provider])
 		for item in items:
 			if item.type != 'item':
 				continue
-			my_data = await self.Libraries[0].read(item.name)
-			if remove_path:
-				assert item.name.startswith(path)
-				tar_name = item.name[len(path):]
-			else:
-				tar_name = item.name
-			info = tarfile.TarInfo(tar_name)
-			my_data.seek(0, io.SEEK_END)
-			info.size = my_data.tell()
-			my_data.seek(0, io.SEEK_SET)
+			data = await provider.read(item.name)
+			if data is None:
+				continue
+			name = item.name[len(path):] if remove_path else item.name
+			info = tarfile.TarInfo(name)
+			data.seek(0, io.SEEK_END)
+			info.size = data.tell()
+			data.seek(0)
 			info.mtime = time.time()
-			tarobj.addfile(tarinfo=info, fileobj=my_data)
+			tarobj.addfile(tarinfo=info, fileobj=data)
+
+		# -- Tenant layers --
```

Comment:
Tenant and global layers will be mixed, so re-import will destroy the correct content of the library.
This is not a correct design.

Reply:
@ateska All the tenant contents will be copied to /tenants/ is that wrong to create the separation between tenant and global?

If so what is the correct approach?

Reply:
Yes - but the export is done for a specific tenant - so only 1 tenant instance will be present in `/tenants/`